ide/nativeimage.api/src/org/netbeans/modules/nativeimage/api/debug/NIDebugger.java

```diff
@@ -165,6 +170,49 @@ public String getVersion() {
         return provider.getVersion();
     }
 
+    /**
+     * Get a list of locations for a given file path.
+     *
+     * @param filePath a file path
+     * @return list of locations, or <code>null</code> when there's no location
+     *         information about such file.
+     * @since 0.2
+     */
+    @CheckForNull
+    public List<Location> listLocations(String filePath) {
+        return provider.listLocations(filePath);
+    }
+
+    /**
+     * Get a list of functions of a given name.
+     *
+     * @param name a name pattern
+     * @param includeNondebug include also symbols from the symbol table
+     * @param maxResults maximum number of results
+     * @return list of source information and their symbols, or <code>null</code>
+     *         when there are no matching symbols.
+     * @since 0.2
+     */
+    @CheckForNull
+    public List<Pair<SourceInfo, List<Symbol>>> listFunctions(String name, boolean includeNondebug, int maxResults) {
+        return provider.listFunctions(name, includeNondebug, maxResults);
+    }
+
+    /**
+     * Get a list of variables of a given name.
+     *
+     * @param name a name pattern
+     * @param includeNondebug include also symbols from the symbol table
+     * @param maxResults maximum number of results
+     * @return list of source information and their symbols, or <code>null</code>
+     *         when there are no matching symbols.
+     * @since 0.2
+     */
+    @CheckForNull
+    public List<Pair<SourceInfo, List<Symbol>>> listVariables(String name, boolean includeNondebug, int maxResults) {
```

Comment:
`List<Pair<X,List<Y>>>` might rather be `Map<X,List<Y>>` - at least that's more traditional way to represent such structure. One can still use `map.entrySet()` and iterate the elements pair by pair (or rather `Entry` by `Entry`).

Reply:
Good point.

Reply:
Done.